benchmarks/benchmark_throughput.py

```diff
@@ -232,14 +246,20 @@ def run_hf(
     use_beam_search: bool,
     max_batch_size: int,
     trust_remote_code: bool,
+    device: str,
 ) -> float:
     assert not use_beam_search
+    is_cuda = device == "cuda"
+    if is_cuda:
+        llm = llm.cuda()
     llm = AutoModelForCausalLM.from_pretrained(
-        model, torch_dtype=torch.float16, trust_remote_code=trust_remote_code)
+        model,
+        torch_dtype=torch.float16 if is_cuda else torch.float32,
+        trust_remote_code=trust_remote_code,
+    )
```

Comment:
load to float32 if not cuda()

https://stackoverflow.com/questions/49995594/half-precision-floating-point-arithmetic-on-intel-chips